packages/libs/eda/src/lib/workspace/ImportAnalysis.tsx

```diff
@@ -44,15 +55,63 @@ export function ImportAnalysis({ analysisClient, analysisId }: Props) {
           };
           history.replace(newLocation);
         },
-        (error) =>
-          setError(error instanceof Error ? error.message : String(error))
+        (error) => {
+          const enforcedErrorObject =
+            error instanceof Error ? error : new Error(String(error));
+          const errorMessageList = enforcedErrorObject.message.split('\n');
+          try {
+            setError(JSON.parse(errorMessageList[1]));
+          } catch {
+            setError(enforcedErrorObject.message);
+          }
+        }
       );
   }, [analysisClient, history, analysisId]);
 
+  const handleClick = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+
+      if (user === undefined || user.isGuest) {
+        dispatch(showLoginForm());
+      }
+    },
+    [history, user, dispatch]
+  );
+
+  const errorMessage =
+    error && typeof error === 'object' ? (
+      error.status === 'forbidden' ? (
+        <>
+          <p>
+            Unable to import this analysis. You do not have access to the study.
+          </p>
+          <p>
+            {user === undefined || user.isGuest ? (
+              <>
+                <button className="link" onClick={handleClick}>
+                  Please login
+                </button>{' '}
+                and try again.
+              </>
+            ) : (
+              <>Please request access from the owner of the study.</>
+            )}
+          </p>
+        </>
+      ) : (
+        <pre>
+          status: {error.status}; message: {error.message}
+        </pre>
+      )
+    ) : (
+      <pre>{error}</pre>
+    );
```

Comment:
**nit**: do you think this file would be improved by factoring this logic into a component?

Since this variable is JSX, we could make this logic more readable by creating a component at the bottom of this file. I personally get lost in nested ternaries. If the type wasn't JSX, I'd put this in an iife. But because we're in JSX land, we can easily factor this out into a component which is the same thing: a function that takes input and returns an output.

```js
function ErrorMessage({ error }) {
    if (!error) return whatever;

    const isErrorObject = typeof error === 'object';
    if (isErrorObject) {
        if (error.status === 'forbidden') {
            return something;
        }
    }

    
    return <pre>{error}</pre>
}
```